website/blog/2021-11-24-2.5.0.md

```diff
@@ -0,0 +1,341 @@
+const f = () => () => (0, 1);
+
+// Prettier 2.4
+const f = () => () => 0, 1;
+
+// Prettier 2.5
+const f = () => () => (0, 1);
+```
+
+#### Ignore errors for sloppy mode syntax ([#11750](https://github.com/prettier/prettier/pull/11750) by [@fisker](https://github.com/fisker), [#11778](https://github.com/prettier/prettier/pull/11778) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+JavaScript’s [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) adds several useful errors to prevent mistakes. Some of these errors are syntax errors that occur at parse time. Since Prettier’s goal is to format all syntactically valid JavaScript code regardless of whether it will actually run, we’ve opted to leave this error checking to linters, compilers, and the runtime.
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+function foo() { var bar = 1; delete bar; }
+
+// Prettier 2.4
+SyntaxError: Deleting local variable in strict mode. (1:31)
+> 1 | function foo() { var bar = 1; delete bar; }
+    |                               ^
+
+// Prettier 2.5
+function foo() {
+  var bar = 1;
+  delete bar;
+}
+```
+
+#### Respect spacing for between expressions and parentheses in embedded CSS ([#11800](https://github.com/prettier/prettier/pull/11800) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+const paragraph2 = css`
+  transform1: ${expr}(30px);
+  transform2: ${expr} (30px);
+`;
+
+// Prettier 2.4
+const paragraph2 = css`
+  transform1: ${expr} (30px);
+  transform2: ${expr} (30px);
+`;
+
+// Prettier 2.5
+const paragraph2 = css`
+  transform1: ${expr}(30px);
+  transform2: ${expr} (30px);
+`;
+
+```
+
+#### Support ES2022 class-private-fields-in syntax in `espree` parser ([#11835](https://github.com/prettier/prettier/pull/11835) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```jsx
+// Example
+class Foo {
+  #brand;
+  static isC(obj) {
+    return #brand in Foo;
+  }
+}
+
+```
+
+### TypeScript
+
+#### Remove unnecessary parentheses for decorators ([#11717](https://github.com/prettier/prettier/pull/11717), [#11849](https://github.com/prettier/prettier/pull/11849) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```tsx
+// Input
+class Test {
+  @foo`bar`
+  test1: string = "test"
+
+  @test().x("global").y()
+  test2: string = "test";
+}
+
+// Prettier 2.4
+class Test {
+  @(foo`bar`)
+  test: string = "test"
+
+  @(test().x("global").y())
+  test2: string = "test";
+}
+
+// Prettier 2.5
+class Test {
+  @foo`bar`
+  test: string = "test"
+
+  @test().x("global").y()
+  test2: string = "test";
+}
+```
+
+### SCSS
+
+#### Improve `@use with` formatting ([#11637](https://github.com/prettier/prettier/pull/11637) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```scss
+// Input
+@use 'library' with (
+  $black: #222,
+  $border-radius: 0.1rem $font-family: 'Helvetica, sans-serif'
```

Comment:
Is this supposed to be invalid Sass? If not, there's a missing comma. (below as well + newline)

Reply:
I think it is valid scss

Reply:
How?

Reply:
I'm not familiar with SCSS, but at least there is a below code on the scss docs.

```scss
@use 'library' with (
  $black: #222,
  $border-radius: 0.1rem
);
```

https://sass-lang.com/documentation/at-rules/use#configuration

Reply:
![image](https://user-images.githubusercontent.com/261509/143453849-eefa436c-1b9f-43dc-9582-8723ef65493f.png)

There are two properties on the same line, with no comma between them.

Reply:
Woops sorry I didn't notice that. I'll fix later, thank you!